### PR TITLE
feat(powered-by): optional server name

### PR DESCRIPTION
**Pull request by PatrickJS**
### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [x] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

**Comment by yusukebe** [comment]
Hi @PatrickJS 

Thank you for the PR. This is good! It is a new *feature* but we can introduce it in the next patch release because it's not such a big change.

There should be tests. Can you add tests?

**Comment by yusukebe** [comment]
@PatrickJS 

I think making the option as an object, not a string, is good. Like the following:

```diff
diff --git a/src/middleware/powered-by/index.ts b/src/middleware/powered-by/index.ts
index caa3be7c..4db425f9 100644
--- a/src/middleware/powered-by/index.ts
+++ b/src/middleware/powered-by/index.ts
@@ -5,9 +5,13 @@

 import type { MiddlewareHandler } from '../../types'

-export const poweredBy = (serverName = 'Hono'): MiddlewareHandler => {
+type PoweredByOptions = {
+  serverName?: string
+}
+
+export const poweredBy = (options?: PoweredByOptions): MiddlewareHandler => {
   return async function poweredBy(c, next) {
     await next()
-    c.res.headers.set('X-Powered-By', serverName)
+    c.res.headers.set('X-Powered-By', options?.serverName ?? 'Hono')
   }
 }
```

**Comment by PatrickJS** [comment]
ok made the changes thanks

**Comment by yusukebe** [comment]
@PatrickJS 

Thanks! Can you add a test?

**Comment by PatrickJS** [comment]
yup just did 🫡

File: src/middleware/powered-by/index.test.ts [hunk 1]
```diff
@@ -11,6 +11,9 @@ describe('Powered by Middleware', () => {
   app.use('/poweredBy2/*', poweredBy())
   app.get('/poweredBy2', (c) => c.text('root'))
 
+  app.use('/poweredByPatrickJS/*', poweredBy({ serverName: 'PatrickJS' }))
```
Comment (yusukebe) [inline src/middleware/powered-by/index.test.ts:14]: Using your name `PatrickJS` is not proper for this test. It's good to use more general words. `foo` is okay though it's boring.

**Comment by PatrickJS** [comment]
done done

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
Thanks @PatrickJS !

Looks good. I'll merge this later and will include this change in the next patch release. It's a new `feat` but it's small. We don't have to wait for the release of the next minor version.

**Pull request merged**